contracts/Collateral.sol

```diff
@@ -5,71 +5,46 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 abstract contract Collateral {
   IERC20 public immutable token;
-  CollateralFunds private _funds;
+  CollateralTotals internal _collateralTotals;
 
   mapping(address => uint256) private _balances;
 
-  constructor(IERC20 token_) collateralInvariant {
```

Comment:
I'm likely missing something, but can we assume that the inputs tested by the fuzzer and the passing result means that we no longer need to enforce parameters using an invariant? Or is there something I am missing where this invariant is allowed to be removed but somehow is still enforced?

Reply:
That's exactly it; by performing enough fuzzing we can gain confidence that the invariants cannot be invalidated, which negates the need for runtime checks. Ideally we would prove that the invariants cannot be invalidated, but that turned out to be problematic: #35